applications/FluidDynamicsApplication/custom_utilities/mass_conservation_utility.cpp

```diff
@@ -30,131 +29,211 @@ namespace Kratos
 /* Public functions *******************************************************/
 
 /// constructor
-MassConservationCheckProcess::MassConservationCheckProcess(
-        ModelPart& rModelPart,
-        const bool PerformCorrections,
-        const int CorrectionFreq,
-        const bool WriteToLogFile,
-        const std::string LogFileName)
-    : Process(), mrModelPart(rModelPart) {
-
-    mCorrectionFreq = CorrectionFreq;
-    mWriteToLogFile = WriteToLogFile;
-    mLogFileName = LogFileName;
-    mPerformCorrections = PerformCorrections;
+MassConservationUtility::MassConservationUtility(
+    Model& rModel,
+    Parameters Settings)
+    : mrModelPart(rModel.GetModelPart(Settings["model_part_name"].GetString())) {
+
+    this->ValidateInputAndInitialize(Settings);
 }
 
 
+
 /// constructor (direct input of settings)
-MassConservationCheckProcess::MassConservationCheckProcess(
+MassConservationUtility::MassConservationUtility(
     ModelPart& rModelPart,
-    Parameters& rParameters)
-    : Process(), mrModelPart(rModelPart) {
+    Parameters Settings)
+    : mrModelPart(rModelPart) {
 
-    Parameters default_parameters( R"(
-    {
-        "model_part_name"                        : "default_model_part_name",
-        "perform_corrections"                    : true,
-        "correction_frequency_in_time_steps"     : 20,
-        "write_to_log_file"                      : true,
-        "log_file_name"                          : "mass_conservation.log"
-    }  )" );
-
-    rParameters.ValidateAndAssignDefaults(default_parameters);
-
-    mCorrectionFreq = rParameters["correction_frequency_in_time_steps"].GetInt();
-    mWriteToLogFile = rParameters["write_to_log_file"].GetBool();
-    mPerformCorrections = rParameters["perform_corrections"].GetBool();
-    mLogFileName = rParameters["log_file_name"].GetString();
+    this->ValidateInputAndInitialize(Settings);
+}
+
+void MassConservationUtility::ValidateInputAndInitialize(
+    Parameters Settings) {
```

Comment:
```suggestion
void MassConservationUtility::ValidateInputAndInitialize(Parameters Settings)
{
```
One argument functions are declared in one line.